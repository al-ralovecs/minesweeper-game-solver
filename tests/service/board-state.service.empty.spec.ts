import { disposition } from '../__fixtures__/state/3x3.empty.json';
import BoardDto from '../../src/dto/board.dto';
import BoardStateService from '../../src/service/board-state.service';

describe('BoardStateService', () => {
    test('check how computes empty board', () => {
        const board = new BoardDto(disposition);
        const boardStateService = new BoardStateService(board.height, board.width, 4);

        boardStateService.setBoard = board;
        boardStateService.process();

        expect(boardStateService.getBoardState).toMatchSnapshot();
    });
});
