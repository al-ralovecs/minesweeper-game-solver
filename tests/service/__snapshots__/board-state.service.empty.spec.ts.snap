// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardStateService check how computes empty board 1`] = `
BoardStateDto {
  "action": Array [
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
  ],
  "actionList": Array [],
  "adjFlagsConfirmed": Array [
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
  ],
  "adjFlagsOnBoard": Array [
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
  ],
  "adjUnrevealed": Array [
    Array [
      3,
      5,
      3,
    ],
    Array [
      5,
      8,
      5,
    ],
    Array [
      3,
      5,
      3,
    ],
  ],
  "adjacentLocations1": Array [
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
  ],
  "board": Array [
    Array [
      -1,
      -1,
      -1,
    ],
    Array [
      -1,
      -1,
      -1,
    ],
    Array [
      -1,
      -1,
      -1,
    ],
  ],
  "chordLocations": Array [],
  "expectedTotalMines": 4,
  "flagConfirmed": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "flagOnBoard": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "height": 3,
  "livingWitnesses": LocationSetDto {
    "data": Array [],
  },
  "numOfHidden": 9,
  "revealed": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "totalFlags": 0,
  "totalFlagsConfirmed": 0,
  "width": 3,
}
`;
